src/Symfony/Component/HttpFoundation/InputBag.php

```diff
@@ -33,7 +33,7 @@ public function get(string $key, mixed $default = null): string|int|float|bool|n
 
         $value = parent::get($key, $this);
 
-        if (null !== $value && $this !== $value && !\is_scalar($value)) {
+        if (null !== $value && $this !== $value && !\is_scalar($value) && !$value instanceof \Stringable) {
```

Comment:
I don't get when it can be an object as the value should come from the PHP global variables?

Reply:
When `InputBag::set()` is called in userland. The `set()` method is public and has explicit `\Stringable` support at the moment, this PR is only makes `get()` consistent

Reply:
I don't get the use case then. I would restrict `set()` instead. But I might be missing something here.

Reply:
I don't get it either, but restricting `set()` should be done on 6.2 with a deprecation as it handles stringable objects since 5.4

Reply:
I would even argue that set() should only support strings.

Reply:
@fabpot in 5.4 only non-scalar values have been deprecated. So in 6.x we still need to support any scalar/stringable object.

Reply:
My point is that I don't see the real-world use case. Here, we should only be dealing with strings as supported by HTTP.

Reply:
Got it for scalars and null: InputBag is hold form data (POST https://github.com/symfony/symfony/pull/40315). I didn't found anything for `\Stringable` objects though, I will open a PR to deprecate it. 
This PR is still relevant as they are supported in versions prior to 6.0.

Reply:
https://github.com/symfony/symfony/pull/34363#discussion_r357991020

i believe it's  mostly to provide a friendly migration path for users, because ppl got used to working with generic ParameterBag

Reply:
> InputBag holds form data

it reflects a structured request body, JSON would fit too

Reply:
Makes sense, thanks.

Reply:
So, if that's temporary, maybe there is no need to add support now to the setter?

Reply:
i'd be in favor of immutability yes :+1: